.github/workflows/process.py

```diff
@@ -0,0 +1,66 @@
+import os
+import os.path as osp
+import json
+
+data = {}
+
+def main():
+    with open("pipelines/pipelines_index.json", 'w') as json_file:
+        check_dir('pipelines')
+        json.dump(data, json_file, ensure_ascii=False)
+
+def check_dir(root_path):
+    for name in os.listdir(root_path):
+        if name == 'README.md' or name == 'pipelines_index.json' or name.startswith('.'):
+            continue
+
+        path = osp.join(root_path, name)
+
+        # Check for other files if this is a directory
+        if osp.isdir(path):
+            check_dir(path)
+            continue
+
+        # Not a directory and not a non-pipeline file
+        # So add this to the data
+        name_no_ext = name.split('.')[0]
+        ext = name.split('.')[1]
+        pipeline = name_no_ext
+        if name_no_ext.endswith('_highres'):
+            pipeline = name_no_ext[:-(len('_highres'))]
+        
+        if not pipeline in data:
+            data[pipeline] = {'json': '', 'screenshots': [], 'metadata': {'userSetData': {}}}
+
+        if ext == 'json':
+            data[pipeline]['json'] = get_direct_url(path)
+            with open(path, 'r') as pipeline_json:
+                pipeline_data = json.load(pipeline_json)
+                old_metadata = data.get(pipeline).get('metadata')
```

Comment:
I see sometimes you use `data.get(pipeline)` and sometimes you use `data[pipeline]`. Maybe stick to `data[pipeline]` style everywhere if there is no special reason to use `data.get`?